cmake/sysbuild/suit.cmake

```diff
@@ -0,0 +1,347 @@
+      list(APPEND STORAGE_BOOT_ARGS
+        --input-envelope ${ROOT_ENVELOPE_SUIT_FILE}
+      )
+  endif()
+
+  sysbuild_get(DEFAULT_BINARY_DIR IMAGE ${DEFAULT_IMAGE} VAR APPLICATION_BINARY_DIR CACHE)
+  # create all storages in the DEFAULT_IMAGE output directory
+  list(APPEND STORAGE_BOOT_ARGS --storage-output-directory "${DEFAULT_BINARY_DIR}/zephyr" --zephyr-base ${ZEPHYR_BASE} ${CORE_ARGS})
+  set_property(
+    GLOBAL APPEND PROPERTY SUIT_POST_BUILD_COMMANDS
+    COMMAND ${PYTHON_EXECUTABLE}
+    ${SUIT_GENERATOR_BUILD_SCRIPT}
+    storage
+    ${STORAGE_BOOT_ARGS}
+  )
+  suit_setup_merge()
+  suit_register_post_build_commands()
+endfunction()
+
+# Setup task to create final and mertged artifact.
+#
+# Usage:
+#   suit_setup_merge()
+#
+function(suit_setup_merge)
+  foreach(image ${IMAGES})
+    set(ARTIFACTS_TO_MERGE)
+
+    sysbuild_get(BINARY_DIR IMAGE ${DEFAULT_IMAGE} VAR APPLICATION_BINARY_DIR CACHE)
+    sysbuild_get(IMAGE_BINARY_DIR IMAGE ${image} VAR APPLICATION_BINARY_DIR CACHE)
+    sysbuild_get(IMAGE_BINARY_FILE IMAGE ${image} VAR CONFIG_KERNEL_BIN_NAME KCONFIG)
+    sysbuild_get(IMAGE_TARGET_NAME IMAGE ${image} VAR CONFIG_SUIT_ENVELOPE_TARGET KCONFIG)
+
+    sysbuild_get(CONFIG_SUIT_ENVELOPE_OUTPUT_ARTIFACT IMAGE ${image} VAR CONFIG_SUIT_ENVELOPE_OUTPUT_ARTIFACT KCONFIG)
+    sysbuild_get(CONFIG_NRF_REGTOOL_GENERATE_UICR IMAGE ${image} VAR CONFIG_NRF_REGTOOL_GENERATE_UICR KCONFIG)
+    sysbuild_get(CONFIG_SUIT_ENVELOPE_OUTPUT_MPI_MERGE IMAGE ${image} VAR CONFIG_SUIT_ENVELOPE_OUTPUT_MPI_MERGE KCONFIG)
+
+    set(OUTPUT_HEX_FILE "${IMAGE_BINARY_DIR}/zephyr/${CONFIG_SUIT_ENVELOPE_OUTPUT_ARTIFACT}")
+
+    list(APPEND ARTIFACTS_TO_MERGE ${BINARY_DIR}/zephyr/storage_${IMAGE_TARGET_NAME}.hex)
+    list(APPEND ARTIFACTS_TO_MERGE ${IMAGE_BINARY_DIR}/zephyr/${IMAGE_BINARY_FILE}.hex)
+    if(CONFIG_SUIT_ENVELOPE_OUTPUT_MPI_MERGE)
+      list(APPEND ARTIFACTS_TO_MERGE ${BINARY_DIR}/zephyr/suit_mpi_${IMAGE_TARGET_NAME}_merged.hex)
+    endif()
+    if(CONFIG_NRF_REGTOOL_GENERATE_UICR)
+      list(APPEND ARTIFACTS_TO_MERGE ${IMAGE_BINARY_DIR}/zephyr/uicr.hex)
+    endif()
+
+    set_property(
+      GLOBAL APPEND PROPERTY SUIT_POST_BUILD_COMMANDS
+      COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/build/mergehex.py
+      --overlap replace
+      -o ${OUTPUT_HEX_FILE}
+       ${ARTIFACTS_TO_MERGE}
+      # fixme: uicr_merged is overwriten by new content, runners_yaml_props_target could be used to define
+      #     what shall be flashed, but not sure where to set this! Remove --overlap if ready!
+      #     example usage: set_property(TARGET runners_yaml_props_target PROPERTY hex_file ${merged_hex_file})
+       COMMAND ${CMAKE_COMMAND} -E copy ${OUTPUT_HEX_FILE} ${IMAGE_BINARY_DIR}/zephyr/uicr_merged.hex
+    )
+  endforeach()
+endfunction()
+
+# Enable SUIT envelope generation only if DEFAULT_IMAGE has it enabled.
+sysbuild_get(CONFIG_SUIT_ENVELOPE IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE KCONFIG)
```

Comment:
Couldn't this be moved inside the if below?

Reply:
`if` below relays on the info extracted here. Can we address that by moving mentioned three functions outside this file - as you proposed, we could suit-utility file containing:
`suit_copy_artifact_to_output_directory`
`suit_render_template`
`suit_create_envelope`